src/components/Kv/KvPhoneInput.vue

```diff
@@ -0,0 +1,201 @@
+						v-for="country in countryList"
+						:key="`country-${country.name}`"
+						:value="country.code"
+					>
+						{{ country.name }}
+					</option>
+				</select>
+			</div>
+
+			<span class="kv-phone-input__country-prefix">
+				+{{ countryCallingCode }}
+			</span>
+
+			<input type="tel"
+				class="kv-phone-input__input"
+				:id="id"
+				:placeholder="placeholderNumber"
+				:value="displayNumber"
+				@input="onInputPhoneNumber"
+			>
+		</div>
+	</div>
+</template>
+
+<script>
+import { getCountryList } from 'flag-icon-css';
+import {
+	AsYouType,
+	getCountryCallingCode,
+	getCountries as getSupportedCountryCodes,
+	getExampleNumber,
+	parsePhoneNumberFromString,
+} from 'libphonenumber-js';
+import exampleNumbers from 'libphonenumber-js/examples.mobile.json'; // used for populating placeholders
+import KvFlag from '@/components/Kv/KvFlag';
+
+const supportedCountryCodes = getSupportedCountryCodes();
+const countryList = getCountryList() // get all country names and codes
+	.filter(countryObj => supportedCountryCodes.includes(countryObj.code)) // remove unsupported countries
+	.sort((a, b) => a.name.localeCompare(b.name)); // alphabetize
+
+export default {
+	components: {
+		KvFlag,
+	},
+	model: {
+		prop: 'value',
+		event: 'input'
+	},
+	props: {
+		id: {
+			type: String,
+			required: true
+		},
+		value: {
+			type: String,
+			default: ''
+		},
+	},
+	data() {
+		return {
+			countryList,
+			displayNumber: this.formatPhoneNumber(this.value), // pretty display of the phone number
+			selectedCountryCode: 'US',
+		};
+	},
+	computed: {
+		countryCallingCode() {
+			return getCountryCallingCode(this.selectedCountryCode);
+		},
+		e164Number() { // E.164 formatted phone number
+			const phoneNumber = parsePhoneNumberFromString(this.displayNumber, this.selectedCountryCode);
+			return phoneNumber?.number || '';
+		},
+		isEmpty() {
+			return this.displayNumber.length === 0;
+		},
+		isValid() {
+			const phoneNumber = parsePhoneNumberFromString(this.displayNumber, this.selectedCountryCode);
+			return phoneNumber?.isValid();
+		},
+		placeholderNumber() {
+			return getExampleNumber(this.selectedCountryCode, exampleNumbers)?.formatNational() || '';
+		},
+	},
+	methods: {
+		formatPhoneNumber(val) {
+			// workaround for https://github.com/catamphetamine/libphonenumber-js/issues/225
+			if (val.includes('(') && !val.includes(')')) {
+				return val.replace('(', '');
+			}
+
+			const asYouTypeFormatter = new AsYouType(this.selectedCountryCode);
+			return asYouTypeFormatter.input(val);
+		},
+		onInputCountry() {
+			this.displayNumber = this.formatPhoneNumber(this.displayNumber);
+			this.emitUpdatedNumber();
+		},
+		onInputPhoneNumber(e) {
+			this.displayNumber = this.formatPhoneNumber(e.target.value);
+			this.emitUpdatedNumber();
+		},
+		emitUpdatedNumber() {
+			/**
+			 * The value of the input in E.164 formatting
+			 * @event input
+			 * @type {Event}
+			 */
+			this.$emit('input', this.e164Number);
```

Comment:
I think to do validation in a parent, we'll need need to emit the selected country as well.

Reply:
Yea I think this could be an object that always includes the `valid` flag as well...

Reply:
If we emit an object, I dont' think we'd be able to do the traditional v-model bindings like 
`<kv-phone-input v-model="myCoolPhoneNumber" />`
We could emit a separate event like 'validity-changed' or something like that though